internal/controller/cache/index/convenience.go

```diff
@@ -87,6 +91,14 @@ var PrefixReferenceOrIDFromArgs = SingleValueFromArgs[resource.ReferenceOrID](fu
 	return PrefixIndexFromRefOrID(r), nil
 })
 
+var MaybePrefixReferenceOrIDFromArgs = SingleValueFromOneOrTwoArgs[resource.ReferenceOrID, IndexQueryOptions](func(r resource.ReferenceOrID, opts IndexQueryOptions) ([]byte, error) {
```

Comment:
This enables all "id" indexes in the cache to support prefix querying on the resource id.